src/game/MatrixGeneration.ts

```diff
@@ -0,0 +1,249 @@
+import GameConfiguration from "./GameConfiguration";
+
+import seedrandom from "seedrandom";
+
+//TODO should this defined here?
+//TODO how to choose these numbers?
+//TODO is this correct way to provide constants? (what are the naming conventions?)
+const MinMatrixSize = 3;
+const MaxMatrixSize = 8;
+
+const MinNumberSequences = 2;
+const MaxNumberSequences = 7;
+const MinSequenceLength = 2;
+
+const MinBufferSize = 4;
+const MaxBufferSize = 8;
+
+const StepsUntilMatrixSizeIncreased = 10;
+const StepsUntilSequenceNumberIncreased = 3;
+
+const DefaultTimeout = 60_000;
+
+//TODO see how many different values are there
+//TODO see how the possible values are defined
+const numberDifferentMatrixValues = 5;
+const matrixValues = ["A0", "E9", "4C", "8B", "6F"];
+const EmptyMatrixValue = "  ";
+
+/**
+ * Generates a randomly filled matrix and according sequences given the level and
+ * a seed.
+ * 
+ * The matrix is always saved in the first array of the function's result.
+ * I.e., result[0] is the matrix, while all other arrays contain sequences.
```

Comment:
Better define and use an interface (like `{matrix: [...], sequences: [...]}`.

Reply:
The code comment is outdated, I updated it: https://github.com/Kugelschieber/breach/pull/22/commits/1bcc038f3b30042bfd05f526d10f2509751699c8
I return the GameConfig interface which was provided by @Ablu . However, it might be useful to discuss if the matrix and sequences should be hard coded to string. I think something more flexible would be nice, e.g. the possible matrix values are contained in an enum and the UI will later decide how to really display these values.